### ISSUE TITLE
[vue/component-options-name-casing]: Feature, accept regular expression

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1908]
**What rule do you want to change?**

vue/component-options-name-casing

**Does this change cause the rule to produce more or fewer warnings?**

Less

**How will the change be implemented? (New option, new default behavior, etc.)?**

Accept regular expression in addition to "PascalCase" | "kebab-case" | "camelCase" options.

**Please provide some example code that this change will affect:**

```vue
  components: {
    "x-field__number": GenericFieldComponent<number>,
    "x-field__string": GenericFieldComponent<string>
  },
```

**What does the rule currently do for this code?**

"_" is not allowed.

**What will the rule do after it's changed?**

Allow "_" with regular expression: `[a-z0-1_-]`

**Additional context**

I am using "x-field__number" naming convention for generic components.

**Comment by ota-meshi** [comment]
Thank you for posting the issue.
However, I don't think the casing you are using is common.
I recommend turning off the rules and writing your own eslint plugin.
Thank you for understanding.
